app/src/org/commcare/activities/RecoveryActivity.java

```diff
@@ -37,132 +37,141 @@
-                                    receiver.displayMessage(Localization.get("auth.over.http"));
-                                }
-                            }
+                                            R.string.recovery_forms_send_failure) :
+                                    StringUtils.getStringRobust(
+                                            RecoveryActivity.this,
+                                            R.string.recovery_forms_send_parital_success,
+                                            String.valueOf(successfulSends));
+                            updateStatus(failureMessage);
+                        } else if (result == FormUploadResult.TRANSPORT_FAILURE) {
+                            updateStatus(R.string.recovery_forms_send_network_error);
+                        } else if (result == FormUploadResult.RECORD_FAILURE) {
+                            updateStatus(Localization.get("sync.fail.individual"));
+                        } else if (result == FormUploadResult.AUTH_OVER_HTTP) {
+                            updateStatus(Localization.get("auth.over.http"));
+                        }
+                    }
 
-                            @Override
-                            protected void deliverUpdate(RecoveryActivity receiver, Long... update) {
-                                //we don't need to deliver updates here, it happens on the notification bar
-                            }
+                    @Override
+                    protected void deliverUpdate(RecoveryActivity receiver, Long... update) {
+                        //we don't need to deliver updates here, it happens on the notification bar
+                    }
 
-                            @Override
-                            protected void deliverError(RecoveryActivity receiver, Exception e) {
-                                Logger.exception("Error in recovery form send: " + ForceCloseLogger.getStackTrace(e), e);
-                                receiver.displayMessage(StringUtils.getStringRobust(receiver, R.string.recovery_forms_send_error) + ": " + e.getMessage());
-                            }
+                    @Override
+                    protected void deliverError(RecoveryActivity receiver, Exception e) {
+                        Logger.exception("Error in recovery form send: " + ForceCloseLogger.getStackTrace(e), e);
+                        updateStatus(StringUtils.getStringRobust(receiver, R.string.recovery_forms_send_error) + ": " + e.getMessage());
+                        loadingIndicator.setVisibility(View.INVISIBLE);
+                    }
 
-                        };
+                };
 
-                mProcess.addSubmissionListener(CommCareApplication.instance().getSession().getListenerForSubmissionNotification());
+        mProcess.addSubmissionListener(CommCareApplication.instance().getSession().getListenerForSubmissionNotification());
+        mProcess.connect(RecoveryActivity.this);

```

Comment:
Hm, isn't this (and executeParallel below) now executable through a common mechanism rather than manually?

Reply:
Got it, sorry! Mis-remembered on my part.